### PR TITLE
MWPW-140336: Refactor region selector and social links decoration for the footer

### PR DESCRIPTION
**Pull request by robert-bogos**
## Description
This solves the issue where the footer is not rendered on blog pages

## Related Issue
Resolves: [MWPW-140336](https://jira.corp.adobe.com/browse/MWPW-140336)

## Test URLs
**Acrobat:**
- Before: https://www.stage.adobe.com/acrobat/online/sign-pdf.html?martech=off
- After: https://www.stage.adobe.com/acrobat/online/sign-pdf.html?martech=off&milolibs=global-footer-decoration--milo--robert-bogos

**BACOM:**
- Before: https://main--bacom--adobecom.hlx.page/products/real-time-customer-data-platform/rtcdp?martech=off
- After: https://main--bacom--adobecom.hlx.page/products/real-time-customer-data-platform/rtcdp?martech=off&milolibs=global-footer-decoration--milo--robert-bogos

**CC:**
- Before: https://main--cc--adobecom.hlx.page/?martech=off
- After: https://main--cc--adobecom.hlx.page/?martech=off&milolibs=global-footer-decoration--milo--robert-bogos

**Milo:**
- Before: https://main--milo--robert-bogos.hlx.page/drafts/rbogos/default?martech=off
- After: https://global-footer-decoration--milo--robert-bogos.hlx.page/drafts/rbogos/default?martech=off

**Homepage:**
- Before: https://main--homepage--adobecom.hlx.page/homepage/index-loggedout?martech=off
- After: https://main--homepage--adobecom.hlx.page/homepage/index-loggedout?martech=off&milolibs=global-footer-decoration--milo--robert-bogos

**Blog:**
- Before: https://main--blog--adobecom.hlx.page/
- After: https://main--blog--adobecom.hlx.page/?milolibs=global-footer-decoration--milo--robert-bogos

### CODE REVIEW

**Review by overmyheadandbody — COMMENTED** [review]
To prevent this from happening up in the future, I'd suggest we also update our _gnav.md_ PR template to include a Blog link

**Review by auniverseaway — APPROVED** [review]

File: libs/blocks/global-footer/global-footer.js [hunk 1]
```diff
@@ -71,17 +71,15 @@ class Footer {
...
-    if (regionAnchor?.href) {
-      regionAnchor.setAttribute('href', `${regionAnchor.getAttribute('href')}#_dnt#_dnb`);
-    }
-    const socialLinks = document.querySelectorAll('.social a');
-    socialLinks.forEach((socialLink) => {
-      socialLink.setAttribute('href', `${socialLink.getAttribute('href')}#_dnb`);
-    });
+    const [region, social] = ['.region-selector', '.social'].map((selector) => this.body.querySelector(selector));
+    const [regionParent, socialParent] = [region?.parentElement, social?.parentElement];
+    [regionParent, socialParent].forEach((parent) => parent?.replaceChildren());
```
Comment (mokimo) [inline libs/blocks/global-footer/global-footer.js:76]: Can we add a comment on why we remove the children and then add them again? It's not obvious when reading this code without prior knowledge

**Review by mokimo — COMMENTED** [review]
The PR title should be equal to the last PRs `Ensure Blog footer is rendered` sounds like it's fixing a really grave (prod) bug

File: libs/blocks/global-footer/global-footer.js [hunk 2]
```diff
@@ -71,17 +71,15 @@ class Footer {
...
+
     decorateLinks(this.body);
 
+    regionParent?.appendChild(region);
+    socialParent?.appendChild(social);
```
[the remark this answers is not in the document]
Comment (robert-bogos) [reply libs/blocks/global-footer/global-footer.js:81]: Done 👍

File: libs/blocks/global-footer/global-footer.js (hunk 1 above)
Comment (robert-bogos) [reply libs/blocks/global-footer/global-footer.js:76]: Done 👍

**Review by mokimo — APPROVED** [review]

**Review by auniverseaway — APPROVED** [review]

**Review by SilviuLCF — APPROVED** [review]
Verified , testing details https://jira.corp.adobe.com/browse/MWPW-140336

**Pull request merged**